src/app/reducers/incidents/incidentSlice.js

```diff
@@ -62,6 +62,55 @@ export const errorLog = state => state.incidents.meta.errorLog;
 export const refreshState = state => state.incidents.meta.refresh;
 // For getting the value of the current state throughout the app;
 
+
+// Formatter Function for preparing selectedIncident Data for DetailsBlock
+// This moves the responsibility out of the component, and into the reducer.
+// Transformation and formatting of data makes more sense in reducer middleware than the React component. (eg: Smart vs Dumb components)
+export const getDetailBlocks = (state) => {
+    const selectedId = selectIncident(state); // Reusing selectors above
+    const incidentList = listIncidents(state); // Reusing selectors above
+    const selectedIncident = incidentList.find((incident) => !incident._id.indexOf(selectedId));
+
+    if (!selectedIncident) { return {}; }
```

Comment:
Returning an empty object instead of `undefined` helps keep redux state from complaining about type errors.